json/src/main/java/io/airlift/json/Codec.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.airlift.json;
+
+public interface Codec<T>
```

Comment:
If you had a method for deserialization here as well, (e.g.: ` public T fromJson(byte[] bytes)`  -> ` public T fromBytes(byte[] bytes)` ) you can nicely abstract a lot of stuff.